### DISCUSSION TITLE
Continuous deconstruction problem about storeToRefs

### DISCUSSION THREAD

**Discussion by Z1oftheKill** [vuejs/pinia#2277] [Help and Questions] [answered]
### What problem is this solving

I defined a store:
`export const useUserStore = defineStore('user', {
  state: () => {
    return {
      userInfo: {
        userName: ''
      }
    }
  },
}`
-----------------------------------------------------------
When I use it:

`import { storeToRefs } from 'pinia'
import { useUserStore } from '@/store'
const { userInfo:{userName} } = storeToRefs(userStore)
`
At this point the `userName`  becomes unresponsive.
What should I do to make it right?

### Proposed solution

I think it can be deconstructed continuously

**Comment by Z1oftheKill** [comment]
One more thing I want to make sure is that if you use storeToRefs you can't use the methods defined in actions?

**Reply by posva** [reply]
You can directly destructure actions: https://pinia.vuejs.org/core-concepts/#using-the-store

**Comment by posva** [comment, marked as answer]
You are losing reactivity by destructuring a primitive from a reactive object (https://vuejs.org/guide/essentials/reactivity-fundamentals.html#reactive-proxy-vs-original-1)

Do

```ts
const {userInfo} = storeToRefs(...)
// or
const store = useStore()
```

And then consume it through the object:

```vue
<p>{{ userInfo.username }}</p>
<p>{{ store.userInfo.username }}</p>
```

I recommend you to give the whole [page](https://vuejs.org/guide/essentials/reactivity-fundamentals.html) a read, it should help in understanding how the reactivity works and why that case breaks. That way you won't fall for it again in the future 😄

**Reply by Z1oftheKill** [reply]
Thank you for your answers and suggestions. I will read the document again carefully
